blazar/db/sqlalchemy/utils.py

```diff
@@ -163,16 +165,51 @@ def get_reservation_allocations_by_host_ids(host_ids, start_date, end_date,
     session = get_session()
     border0 = models.Lease.end_date < start_date
     border1 = models.Lease.start_date > end_date
-    query = (session.query(models.Reservation, models.ComputeHostAllocation)
-             .join(models.Lease).join(models.ComputeHostAllocation)
-             .filter(models.ComputeHostAllocation.compute_host_id
-                     .in_(host_ids))
-             .filter(~sa.or_(border0, border1)))
+    fields = ['id', 'status', 'lease_id', 'start_date',
+              'end_date', 'lease_name', 'project_id']
+
+    reservations_query = (session.query(
+        models.Reservation.id,
+        models.Reservation.status,
+        models.Reservation.lease_id,
+        models.Lease.start_date,
+        models.Lease.end_date,
+        models.Lease.name,
+        models.Lease.project_id)
+        .join(models.Lease)
+        .filter(models.Reservation.deleted.is_(None))
+        .filter(sa.or_(
+            models.Lease.status.like(status.lease.ACTIVE),
+            models.Lease.status.like(status.lease.PENDING)))
+        .filter(sa.or_(
+            models.Reservation.status.like(status.reservation.ACTIVE),
+            models.Reservation.status.like(status.reservation.PENDING)))
+        .filter(~sa.or_(border0, border1)))
+
     if lease_id:
-        query = query.filter(models.Reservation.lease_id == lease_id)
+        reservations_query = reservations_query.filter(
+            models.Reservation.lease_id == lease_id)
     if reservation_id:
-        query = query.filter(models.Reservation.id == reservation_id)
-    return query.all()
+        reservations_query = reservations_query.filter(
+            models.Reservation.id == reservation_id)
+
+    reservations = [dict(zip(fields, r)) for r in reservations_query.all()]
```

Comment:
This is a very clever use of `zip`! I wouldn't have thought it could work that way.